includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -438,6 +438,10 @@ public function cancelled_subscription( $subscription ) {
 
 		$charge_permission_id = $subscription->get_meta( 'amazon_charge_permission_id' );
 
+		if ( empty( $charge_permission_id ) ) {
+			return;
+		}
+
```

Comment:
The return here is because in some cases, the meta doesn't exist, therefore there's nothing to cancel or log a change for.

The actual status change is not done in this function, this is used to cancel the ChargePermission on amazon